### PR TITLE
[rv_timer/top] Remove rv_timer from PLIC sources

### PR DESCRIPTION
**Pull request by msfschaffner**
The RV_TIMER interrupt is already connected to a dedicated RV_CORE_IBEX interrupt port, hence the connection to RV_PLIC is redundant and can be removed.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/top_earlgrey/data/top_earlgrey.hjson [hunk 1]
```diff
@@ -1058,6 +1058,39 @@
...
+    "aon_timer_aon",
+    "flash_ctrl",
+    "hmac",
+    "kmac",
+    "keymgr",
+    "csrng",
+    "entropy_src",
+    "edn0",
+    "edn1",
+    "otbn"
```
Comment (eunchan) [inline hw/top_earlgrey/data/top_earlgrey.hjson:1091]: LGTM. Maybe we should consider `exclude` option as well.

**Review by eunchan — APPROVED** [review]

**Review by sriyerg — APPROVED** [review]

**Comment by sriyerg** [comment]
CI fails because the SW tests are now broken as expected.

File: hw/top_earlgrey/data/top_earlgrey.hjson (hunk 1 above)
Comment (timothytrippel) [reply hw/top_earlgrey/data/top_earlgrey.hjson:1091]: LGTM,  @sriyerg, @moidx, and I were discussing this yesterday, as we are looking to port FreeRTOS to OT for driving system level (concurrency) tests (we noticed the timer interrupts had two paths to Ibex). Nice catch!

**Comment by timothytrippel** [comment]
LGTM,  @sriyerg, @moidx, and I were discussing this yesterday, as we are looking to port FreeRTOS to OT for driving system level (concurrency) tests (we noticed the timer interrupts had two paths to Ibex). Nice catch!

**Comment by jettr** [comment]
I just saw this now, and I was hoping that we could still have the timer interrupts appear in the plic as well. We will most likely have multiple timers going (i.e. RTC and other main timer), and it is nice to be able to differentiate those two sources with a different IRQ number instead of polling both timers after the main `mtimer` bit is set in the CSR

**Comment by tjaychen** [comment]
Hmmm... that's a good point i had not thought about.
Right now we have rv_timer and aon_timer (the latter is basically like an
rtc).

The latter is hooked into the plic, while the former goes into the ibex
core directly.

**Comment by ccteng-google** [comment]
Does that mean rv_timer only triggers mtimer interrupt in the top half? And aon_timer triggers mext?

**Comment by tjaychen** [comment]
yes i believe that's true.

**Comment by msfschaffner** [comment]
@ccteng-google, @jettr  Is the `mtimer` / `mext` separation sufficient for your purposes? 
If not, we can revert this change to give SW more flexibility...

**Comment by tjaychen** [comment]
i feel like this doesn't need reverting...the worst case i think is that we need to somehow combine aon_timer / rv_timer into `mtimer`.  I don't really see a reason for that though, since it's not like `aon_timer` is standardized like `rv_timer`.  To me it's okay for `aon_timer` to go into peripheral interrupts. 

Hope everyone here agrees...

**Comment by ccteng-google** [comment]
I'd like to better understand re:
1. What's the benefit/advantage of having rv_timer -> mtimer and aon_timer -> mext?
2. Is there any drawback/disadvantage or handling both timers as mext? Or is that even possible?

**Comment by jettr** [comment]
Do we think  we will ever support two 64-bit timers (in addition to aon)? We have used other silicon with multiple main timers in the past and each of those timers had their own `mext` IRQ number. If we ever plan on that being a possibility, what would we do in that case? (we could have each timer have an IRQ and the `mtimer` bit could be a logical or of the two  --- or give `mtimer` to one of the timers and the other timer is only in the plic just like aon)

The advantage of at least one of the timers firing `mtimer` is that we are complying with RISC-V standard

Thinking about this more, having `mtimer` dedicated to only one hardware timer and the other hardware timers getting routed through `mext` IRQs seems like enough for software to differentiate easily.

However, another consideration is that `mext` takes priority over `mtimer` if I recall the RISC-V specification correctly (I can't find a reference to that though).  There may be an application where getting the timer interrupt is more important than handling i2c transactions. That may be reason enough to dual route the timer to `mtimer` (for RISC-V compliance) and to `mext` (to give priority flexibility)

**Comment by ccteng-google** [comment]
> Thinking about this more, having mtimer dedicated to only one hardware timer and the other hardware timers getting routed through mext IRQs seems like enough for software to differentiate easily.

Currently there is only one 64-bit timer that goes to `mtimer`. How about we start by adding a 64-bit timer (with two compares) to `mext'?

**Comment by tjaychen** [comment]
the timer feature set is closed for this revision.
if we want to add more timer features, please file a separate issue.

**Comment by msfschaffner** [comment]
Reverting the commit in this PR is however possible, since it is not a `rv_timer` block-level change.
Done here: #7509

**Pull request merged**